rust/src/io/object_store.rs

```diff
@@ -49,6 +48,20 @@ pub struct ObjectStore {
     prefetch_size: usize,
 }
 
+/// URI schemes supported by lance
+mod schemes {
```

Comment:
We dont need a mod for this purpose?  Feel that it is a misuse of mod (to pretend to be a enum).